process/smartContract/processorV2/processV2.go

```diff
@@ -939,20 +993,22 @@ func (sc *scProcessor) doExecuteBuiltInFunction(
 				tx:                   tx,
 				scrTxs:               scrResults,
 				acntSnd:              acntSnd,
-				snapshot:             snapshot,
 				createdAsyncCallback: createdAsyncCallback,
+				failureContext:       failureContext,
 			})
 		if errReturnCode != vmcommon.Ok || err != nil {
-			return errReturnCode, sc.ProcessIfError(acntSnd, txHash, tx, err.Error(), []byte(err.Error()), snapshot, vmInput.GasLocked)
+			return errReturnCode, nil
 		}
 	}
 
-	errCheck := sc.checkSCRSizeInvariant(scrResults)
-	if errCheck != nil {
-		return vmcommon.UserError, sc.ProcessIfError(acntSnd, txHash, tx, errCheck.Error(), []byte(errCheck.Error()), snapshot, vmInput.GasLocked)
+	err = sc.checkSCRSizeInvariant(scrResults)
+	if err != nil {
+		failureContext.setMessagesFromError(err)
+		return vmcommon.UserError, nil
 	}
 
-	return sc.finishSCExecution(scrResults, txHash, tx, newVMOutput, builtInFuncGasUsed)
+	returnCode, err = sc.finishSCExecution(scrResults, txHash, tx, newVMOutput, builtInFuncGasUsed)
```

Comment:
why these changes?

Reply:
+1

Reply:
at some point there was some code between that and return :) ofc, fixed